File: templates/contracts/js/src/contract.ts
// Find all our documentation at https://docs.near.org
import { NearBindgen, near, call, view } from 'near-sdk-js';

@NearBindgen({})
class HelloNear {
  message: string = "Hello";

  @view({}) // This method is read-only and can be called for free
  get_greeting(): string {
    return this.message;
  }

  @call({}) // This method changes the state, for which it cost gas
  set_greeting({ message }: { message: string }): void {
    near.log(`Saving greeting ${message}`);
    this.message = message;
  }
}

File: templates/contracts/py/contract.py
from near_sdk_py import Contract, view, call, init

class GreetingContract(Contract):
    """
    A simple greeting contract that stores and returns a message.
    """
    
    @init
    def initialize(self, default_message="Hello, NEAR world!"):
        """
        Initialize the contract with a default greeting message.
        """
        self.storage["greeting"] = default_message
        return {"success": True}
    
    @call
    def set_greeting(self, message: str):
        """
        Change the greeting message.
        """
        self.storage["greeting"] = message
        self.log_info(f"Saving greeting: {message}")
        return {"success": True}
    
    @view
    def get_greeting(self):
        """
        Retrieve the current greeting message.
        """
        return self.storage.get("greeting", "Hello, NEAR world!")
        